plugins/entity-feedback/src/components/FeedbackResponseDialog/FeedbackResponseDialog.tsx

```diff
@@ -81,13 +87,41 @@ export const FeedbackResponseDialog = (props: FeedbackResponseDialogProps) => {
   const classes = useStyles();
   const errorApi = useApi(errorApiRef);
   const feedbackApi = useApi(entityFeedbackApiRef);
+  const alertApi = useApi(alertApiRef);
   const [responseSelections, setResponseSelections] = useState(
     Object.fromEntries(feedbackDialogResponses.map(r => [r.id, false])),
   );
   const [comments, setComments] = useState('');
   const [consent, setConsent] = useState(true);
+  const requireComments = feedbackDialogResponses
+    .filter(r => r.mustComment)
+    .map(r => r.id);
+
+  const isMandatedBoxChecked = () => {
+    const checkedBoxes = Object.keys(responseSelections).filter(
+      id => responseSelections[id],
+    );
+    return checkedBoxes.some(id => requireComments.includes(id));
+  };
 
   const [{ loading: saving }, saveResponse] = useAsyncFn(async () => {
+    if (requireComments.length > 0) {
+      if (comments.length === 0 && isMandatedBoxChecked()) {
+        alertApi.post({
+          message:
+            'The selected option(s) require a comment. Please provide a comment.',
+          severity: 'info',
+        });
+        return;
+      }
+      if (comments.length > 0 && !isMandatedBoxChecked()) {
+        alertApi.post({
+          message: 'Please select the option(s) that require a comment.',
```

Comment:
Can we leave this as an option to pass in (with `mustComment`) but remove it from the default `feedbackDialogResponses`?

Reply:
sure I think a option to enable this works - apart from adding this option, it might be more intuitive that if this is enabled, instead of producing this alert, have the UI disable the comment box unless a comment required option is selected (and ignore the comment if those options are unselected). That way you would never get into this scenario

Reply:
Hi @kuangp I reviewed your feedback with our design lead & she agreed to make the following changes:
- remove link between comment text and box selection
- remove alerts
- add a comment box for each option where `mustComment` is `true`, this comment box should be hidden until the user checks this box. Filling out the comment box should be optional. 
- the Feedback response table should have some tie between comments & responses
Should I mark this PR as `Draft` until I can get those changes up?

Reply:
Hi @kuangp just FYI, our design team provided specks with the updated suggestions. I'm including it as a visual to what I described in my comment last week
![Screenshot 2024-02-13 at 10 39 46 AM](https://github.com/backstage/backstage/assets/146387693/7ce6ac24-b4b1-486d-9342-b07bf424b777)